Index: ukui-settings-daemon-1.2.1/plugins/xsettings/usd-xsettings-manager.c
===================================================================
--- ukui-settings-daemon-1.2.1.orig/plugins/xsettings/usd-xsettings-manager.c	2020-05-22 11:18:45.000000000 +0800
+++ ukui-settings-daemon-1.2.1/plugins/xsettings/usd-xsettings-manager.c	2020-06-29 10:22:27.177372551 +0800
@@ -466,6 +466,20 @@
 		tmpCursorSize = XcursorGetDefaultSize(dpy);
 	}
 	// end add by liutong 
+    
+    /* 添加DPI缩放至文件
+     * Add DPI zoom to file */
+    gchar    *XreFilePath;
+    gchar    *date;
+    gboolean  res;
+    XreFilePath = g_build_filename (g_get_home_dir (), ".Xresources", NULL);
+    date = g_malloc (40);
+    g_sprintf(date,"Xft.dpi:%d\nXcursor.size:%d",
+                    settings->dpi/1024, settings->cursor_size);
+    res = g_file_set_contents(XreFilePath,date,strlen(date),NULL);
+    if(!res)
+        g_debug("Xresources File write failed ");
+    /* end add by Shang Xiaoyang */
 
         update_property (add_string, "Xft.dpi",
                                 g_ascii_dtostr (dpibuf, sizeof (dpibuf), (double) settings->dpi / 1024.0));
