Index: ukui-settings-daemon-1.2.1/plugins/background/usd-background-manager.c
===================================================================
--- ukui-settings-daemon-1.2.1.orig/plugins/background/usd-background-manager.c
+++ ukui-settings-daemon-1.2.1/plugins/background/usd-background-manager.c
@@ -192,14 +192,12 @@ static void
 real_draw_bg (UsdBackgroundManager *manager,
 	      GdkScreen		   *screen)
 {
-/*
 	UsdBackgroundManagerPrivate *p = manager->priv;
 	GdkWindow *window = gdk_screen_get_root_window (screen);
 	gint width   = gdk_screen_get_width (screen);
 	gint height  = gdk_screen_get_height (screen);
 
 	free_bg_surface (manager);
-//modify begin by liutong
 	p->surface = mate_bg_create_surface (p->bg, window, width, height, TRUE);
 	if (p->do_fade)
 	{
@@ -212,9 +210,7 @@ real_draw_bg (UsdBackgroundManager *mana
 		mate_bg_set_surface_as_root (screen, p->surface);
 	}
 
-//modify end by liutong
 	p->scr_sizes = g_list_prepend (p->scr_sizes, g_strdup_printf ("%dx%d", width, height));
-*/
 }
 
 static void
@@ -519,6 +515,10 @@ usd_background_manager_start (UsdBackgro
 			  G_CALLBACK (on_bg_handling_changed), manager);
 	g_signal_connect (p->settings, "changed::" MATE_BG_KEY_SHOW_DESKTOP,
 			  G_CALLBACK (on_bg_handling_changed), manager);
+	g_signal_connect (p->settings, "changed::" MATE_BG_KEY_PICTURE_FILENAME,
+			  G_CALLBACK (on_bg_handling_changed), manager);
+	g_signal_connect (p->settings, "changed::" MATE_BG_KEY_PRIMARY_COLOR,
+			  G_CALLBACK (on_bg_handling_changed), manager);
 
 	/* If Peony is set to draw the background, it is very likely in our session.
 	 * But it might not be started yet, so peony_is_drawing_bg() would fail.
Index: ukui-settings-daemon-1.2.1/plugins/xsettings/usd-xsettings-manager.c
===================================================================
--- ukui-settings-daemon-1.2.1.orig/plugins/xsettings/usd-xsettings-manager.c
+++ ukui-settings-daemon-1.2.1/plugins/xsettings/usd-xsettings-manager.c
@@ -577,10 +577,11 @@ xft_settings_set_xresources (UkuiXftSett
     gchar    *date;
     gboolean  res;
     XreFilePath = g_build_filename (g_get_home_dir (), ".Xresources", NULL);
-    date = g_malloc (40);
-    g_sprintf(date,"Xft.dpi:%d\nXcursor.size:%d",
+    date = g_malloc (128);
+    g_sprintf(date,"Xft.dpi:%d\nXcursor.size:%d\nXcursor.theme:%s\n",
                     settings->scaled_dpi/1024, 
-                    settings->cursor_size);
+                    settings->cursor_size,
+                    tmpCursorTheme);
 
     res = g_file_set_contents(XreFilePath,date,strlen(date),NULL);
     if(!res)
