Index: ukui-settings-daemon-1.2.1/plugins/media-keys/usd-media-keys-manager.c
===================================================================
--- ukui-settings-daemon-1.2.1.orig/plugins/media-keys/usd-media-keys-manager.c	2020-05-22 11:18:45.000000000 +0800
+++ ukui-settings-daemon-1.2.1/plugins/media-keys/usd-media-keys-manager.c	2020-06-12 11:36:44.556079529 +0800
@@ -981,19 +981,19 @@
 static void
 do_screenshot_action (UsdMediaKeysManager *manager)
 {
-        execute (manager, "gnome-screenshot",FALSE,FALSE);
+        execute (manager, "mate-screenshot",FALSE,FALSE);
 }
 
 static void
 do_area_screenshot_action (UsdMediaKeysManager *manager)
 {
-        execute (manager, "gnome-screenshot -a",FALSE,FALSE);
+        execute (manager, "mate-screenshot -a",FALSE,FALSE);
 }
 
 static void
 do_window_screenshot_action (UsdMediaKeysManager *manager)
 {
-        execute (manager, "gnome-screenshot -w",FALSE,FALSE);
+        execute (manager, "mate-screenshot -w",FALSE,FALSE);
 }
 
 
