Index: ukui-settings-daemon-1.2.1/plugins/keybindings/usd-keybindings-manager.c
===================================================================
--- ukui-settings-daemon-1.2.1.orig/plugins/keybindings/usd-keybindings-manager.c	2020-05-22 11:18:45.000000000 +0800
+++ ukui-settings-daemon-1.2.1/plugins/keybindings/usd-keybindings-manager.c	2020-08-12 09:58:58.411776474 +0800
@@ -38,6 +38,7 @@
 #include <gtk/gtk.h>
 #include <X11/keysym.h>
 #include <gio/gio.h>
+#include <gio/gdesktopappinfo.h>
 #include <dconf.h>
 
 #include "ukui-settings-profile.h"
@@ -47,6 +48,7 @@
 #include "usd-keygrab.h"
 #include "eggaccelerators.h"
 
+#define DESKTOP_APP_DIR "/usr/share/applications/"
 #define GSETTINGS_KEYBINDINGS_DIR "/org/ukui/desktop/keybindings/"
 #define CUSTOM_KEYBINDING_SCHEMA "org.ukui.control-center.keybinding"
 
@@ -485,7 +487,8 @@
                         }
 
                         envp = get_exec_environment (xevent);
-
+                        
+                        /*
                         retval = g_spawn_async (NULL,
                                                 argv,
                                                 envp,
@@ -494,6 +497,13 @@
                                                 NULL,
                                                 NULL,
                                                 &error);
+                        */
+
+                        char execPathName[255];
+                        sprintf(execPathName, "%s%s", DESKTOP_APP_DIR, binding->action);
+                        GDesktopAppInfo *info = g_desktop_app_info_new_from_filename(execPathName);
+                        retval = g_app_info_launch_uris(info, NULL, NULL, NULL);
+
                         g_strfreev (argv);
                         g_strfreev (envp);
 
