Index: ukui-settings-daemon-1.2.1/plugins/xrandr/usd-xrandr-manager.c
===================================================================
--- ukui-settings-daemon-1.2.1.orig/plugins/xrandr/usd-xrandr-manager.c	2020-07-06 20:54:55.000000000 +0800
+++ ukui-settings-daemon-1.2.1/plugins/xrandr/usd-xrandr-manager.c	2020-07-31 14:16:59.217589366 +0800
@@ -41,6 +41,9 @@
 #include <gio/gio.h>
 #include <dbus/dbus-glib.h>
 
+#include <X11/Xlib.h>
+#include <X11/extensions/XInput2.h>
+
 #define MATE_DESKTOP_USE_UNSTABLE_API
 #include <libmate-desktop/mate-rr-config.h>
 #include <libmate-desktop/mate-rr.h>
@@ -1708,6 +1711,68 @@
         }
 }
 
+/*检测到旋转，更改触摸屏鼠标光标位置*/
+void set_touchscreen_cursor(void *date)
+{
+        Display *xdisplay = XOpenDisplay(NULL);
+        Atom property_atom;
+        property_atom = XInternAtom (xdisplay, "Coordinate Transformation Matrix", True);
+        if (!property_atom)
+            return;
+        Atom type = XInternAtom (xdisplay, "FLOAT", False);
+        XIChangeProperty (xdisplay, XITouchClass, property_atom, type,
+                          32, XIPropModeReplace, date, 9);
+        XCloseDisplay(xdisplay);
+}
+
+/* 设置触摸屏触点的角度 */
+void set_touchscreen_cursor_rotation(MateRRScreen *screen)
+{
+        /* 添加触摸屏鼠标设置 */
+        MateRRConfig *result;
+        MateRROutputInfo **outputs;
+        int i;
+        result = mate_rr_config_new_current (screen, NULL);
+        outputs = mate_rr_config_get_outputs (result);
+        for(i=0; outputs[i] != NULL;++i)
+        {
+            MateRROutputInfo *info = outputs[i];
+            if(mate_rr_output_info_is_connected (info)){
+                int rotation = mate_rr_output_info_get_rotation(info);
+                switch(rotation){
+                    case MATE_RR_ROTATION_0:
+                    {
+                        float full_matrix[9] = {1, 0, 0, 0, 1, 0, 0, 0, 1};
+                        set_touchscreen_cursor(&full_matrix);
+                        break;
+                    }
+                    case MATE_RR_ROTATION_90:
+                    {
+                        float full_matrix[9] = {0, -1, 1, 1, 0, 0, 0, 0, 1};
+                        set_touchscreen_cursor(&full_matrix);
+                        break;
+                    }
+                    case MATE_RR_ROTATION_180:
+                    {
+                        float full_matrix[9] = {-1,  0, 1, 0, -1, 1, 0, 0, 1};
+                        set_touchscreen_cursor(&full_matrix);
+                        break;
+                    }
+                    case MATE_RR_ROTATION_270:
+                    {
+                        float full_matrix[9] = {0, 1, 0, -1, 0, 1, 0, 0, 1};
+                        set_touchscreen_cursor(&full_matrix);
+                        break;
+                    }
+                    default:
+                    {
+                        float full_matrix[9] = {1, 0, 0, 0, 1, 0, 0, 0, 1};
+                        set_touchscreen_cursor(&full_matrix);
+                    }
+                }
+            }
+        }
+}
 
 static void
 on_randr_event (MateRRScreen *screen, gpointer data)
@@ -1781,6 +1846,9 @@
                 } else
                         log_msg ("Applied stored configuration to deal with event\n");
         }
+        
+        /* 添加触摸屏鼠标设置 */
+        set_touchscreen_cursor_rotation(screen);
 
         /* poke mate-color-manager */
         apply_color_profiles ();
@@ -2576,6 +2644,10 @@
         gdk_window_add_filter (gdk_get_default_root_window(),
                                (GdkFilterFunc)event_filter,
                                manager);
+        
+
+        /* 添加触摸屏鼠标设置 */
+        set_touchscreen_cursor_rotation(manager->priv->rw_screen);
 
         start_or_stop_icon (manager);
 
